[shopsys] move data fixtures to project-base (#854)
shopsys/shopsys: docs/cookbook/adding-new-attribute-to-an-entity.md

```

## Data fixtures
You can modify data fixtures as they are located in ShopBundle.
### 1. Random `extId`

> If you want to add unique random `extId` every product, you can add the generation to function `createProductDataFromRowForFirstDomain` in [`ProductDataFixtureLoader.php`](../../project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/ProductDataFixtureLoader.php).

Author: I changed the sentence so it's more understandable

+    * @param \Faker\Generator $faker  
     */
    public function __construct(
        ProductParametersFixtureLoader $productParametersFixtureLoader,
        ProductDataFactoryInterface $productDataFactory,
        Domain $domain,
        PricingGroupFacade $pricingGroupFacade,
+       Faker $faker
    ) {
        $this->productParametersFixtureLoader = $productParametersFixtureLoader;
        $this->productDataFactory = $productDataFactory;
        $this->domain = $domain;
        $this->pricingGroupFacade = $pricingGroupFacade;
+       $this->faker = $faker;  
    }

    //...

    public function createProductDataFromRowForFirstDomain($row)
        {
            $productData = $this->productDataFactory->create();
            $this->updateProductDataFromCsvRowForFirstDomain($productData, $row);
+           $productData->extId = $this->faker->unique()->numberBetween(1, 10000);

            return $productData;
        }
```

### 2. Specific `extId`
If you need to add specific `extId` to products in data fixture you can add new column to [`demo-data-products.csv`](../../project-base/src/Shopsys/ShopBundle/DataFixtures/resources/demo-data-products.csv).

> Then you have add the setting of the value to `updateProductDataFromCsvRowForFirstDomain` in [`ProductDataFixtureLoader.php`](../../project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/ProductDataFixtureLoader.php).

Reviewer: could be like this? or is it about something else?
```suggestion
Then you have to set the value of the new column in `updateProductDataFromCsvRowForFirstDomain` method of [`ProductDataFixtureLoader.php`](../../project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/ProductDataFixtureLoader.php).
```

```diff
+   const COLUMN_EXT_ID = 24;  

    //...

> 

Reviewer: too many white lines for me BUT OK

    /**
     * @param \Shopsys\FrameworkBundle\Model\Product\ProductData $productData
     * @param array $row
     */
    protected function updateProductDataFromCsvRowForFirstDomain(ProductData $productData, array $row)
    {

        //...

+       $productData->extId = $row[self::COLUMN_EXT_ID];
    }  
```

> 

Reviewer: remove last line please

```

## Data fixtures
<!--- TODO during release exchange this sentence with examples below --->
Currently, it is not possible to modify data fixtures from your project, this issue will be addressed in near future.

> <!--- TODO

Author: remove comment

```

## Data fixtures

> You can modify data fixtures as they are located in project-base.

Reviewer: ```suggestion
You can modify data fixtures in `src/Shopsys/ShopBundle/DataFixtures/` of your project.
```
i am not sure about this one, maybe it should be decided during BV

Author: I'll discuss it during BV

+       $productData->extId = $row[self::COLUMN_EXT_ID];
    }  
```

> ---!>

Reviewer: ???? --!> ????
